[2025-02-09T12:41:59.889+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-02-09T12:41:59.942+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_MusicStreamPipeline.MusicStream_delete manual__2025-02-09T12:41:58.519716+00:00 [queued]>
[2025-02-09T12:42:00.780+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_MusicStreamPipeline.MusicStream_delete manual__2025-02-09T12:41:58.519716+00:00 [queued]>
[2025-02-09T12:42:00.781+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 1
[2025-02-09T12:42:00.810+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): MusicStream_delete> on 2025-02-09 12:41:58.519716+00:00
[2025-02-09T12:42:00.827+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=1088) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-02-09T12:42:00.825+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'etl_MusicStreamPipeline', 'MusicStream_delete', 'manual__2025-02-09T12:41:58.519716+00:00', '--job-id', '541', '--raw', '--subdir', 'DAGS_FOLDER/etl_dag.py', '--cfg-path', '/tmp/tmpy8ezlz0w']
[2025-02-09T12:42:00.829+0000] {standard_task_runner.py:63} INFO - Started process 1107 to run task
[2025-02-09T12:42:00.835+0000] {standard_task_runner.py:91} INFO - Job 541: Subtask MusicStream_delete
[2025-02-09T12:42:00.934+0000] {task_command.py:426} INFO - Running <TaskInstance: etl_MusicStreamPipeline.MusicStream_delete manual__2025-02-09T12:41:58.519716+00:00 [running]> on host e231ebee4370
[2025-02-09T12:42:01.104+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Le Huynh Thanh Duong' AIRFLOW_CTX_DAG_ID='etl_MusicStreamPipeline' AIRFLOW_CTX_TASK_ID='MusicStream_delete' AIRFLOW_CTX_EXECUTION_DATE='2025-02-09T12:41:58.519716+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-02-09T12:41:58.519716+00:00'
[2025-02-09T12:42:01.106+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-02-09T12:42:01.219+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-02-09T12:42:01.221+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/etls/load.py", line 12, in drop_table
    cur.execute(sql)
psycopg2.errors.ForeignKeyViolation: update or delete on table "dim_artists" violates foreign key constraint "dim_songs_artist_key_fkey" on table "dim_songs"
DETAIL:  Key (artist_key)=(2) is still referenced from table "dim_songs".

[2025-02-09T12:42:01.241+0000] {taskinstance.py:1205} INFO - Marking task as FAILED. dag_id=etl_MusicStreamPipeline, task_id=MusicStream_delete, execution_date=20250209T124158, start_date=20250209T124159, end_date=20250209T124201
[2025-02-09T12:42:01.274+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 541 for task MusicStream_delete (update or delete on table "dim_artists" violates foreign key constraint "dim_songs_artist_key_fkey" on table "dim_songs"
DETAIL:  Key (artist_key)=(2) is still referenced from table "dim_songs".
; 1107)
[2025-02-09T12:42:01.293+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-02-09T12:42:01.313+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-09T12:42:01.315+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
